lib/handler.js

```diff
@@ -68,6 +70,13 @@ const handler = async (req, res, pgPool) => {
       res,
       pgPool,
       fetchMinersRSRSummary)
+  } else if (req.method === 'GET' && segs[0] === 'nodes' && segs[1] === 'daily' && segs.length === 2) {
```

Comment:
Proposed solution for extracting the platform-specific routes:

```js
  // at the top
  import { handlePlatformRoute } from './platform-routes.js'

  // inside the handler
  if (req.method === 'GET' && segs[0] === 'retrieval-success-rate' && segs.length === 1) {
  // ... (more routes) ...
  } else if (req.method === 'GET' && segs.length === 0) {
    // health check - required by Grafana datasources
    res.end('OK')
  } else if (await handlePlatformRoute(req, res, pgPool) {
    // no-op, request was handled
  } else 
    notFound(res)
  }
```

Reply:
Sounds good! Would you also prefer that we duplicate the getStatsWithFilterAndCaching function in platform-routes.js?

We could alternatively pass the function in as a parameter to handlePlatformRoute, and when we port to Voyager, we just copy it from handler.js and remove as a handlePlatformRoute parameter.

Up to you!

Reply:
I think `getStatsWithFilterAndCaching` should have the same implementation in Spark and Voyager. In that case, moving the implementation to `platform-routes.js` and letting the project-specific handler file import it from there will require the least amount of maintenance.

Alternatively, and that's an even cleaner option, can we move `getStatsWithFilterAndCaching` and its helpers into its own file?

Let's implement the second option if you agree.